### PR TITLE
Add support for querying CRDTs with Yokozuna

### PR DESCRIPTION
**Pull request by bkerley**
Existing:

```ruby
client.search 'sets2', 'set:asdfasdfasd'
# returns:
{"max_score"=>0.712317943572998,
 "num_found"=>1,
 "docs"=>
  [{"score"=>"7.12317940000000038303e-01",
    "_yz_rb"=>"bryce",
    "_yz_rt"=>"sets",
    "_yz_rk"=>"asdf",
    "_yz_id"=>"1*sets*bryce*asdf*52"}]}
```

Desired:

```ruby
idx = client.index 'sets2'
results = idx.search 'set:asdfasdfasd'
# results is an array-like collection
results.first #=> the actual set
results.docs.first #=> the docs entry
```

Looks figuring out that the results are a set type either involves probing an entry's content type (which breaks if a user lies and says something is a crdt when it isn't), or adding an API to read bucket type properties.

### CODE REVIEW

File: lib/riak/crdt/base.rb [hunk 1]
```diff
@@ -61,6 +61,14 @@ def context?
         !!@context
       end
 
+      def ==(other)
+        return false unless other.is_a? Riak::Crdt::Base
```
Comment (bkerley) [inline lib/riak/crdt/base.rb]: Should probably also Care about matching classes between self and other.

**Comment by christophermancini** [comment]
Tests pass. Code looks clean. Looks like there is good test coverage.

**Comment by christophermancini** [comment]
:+1: 772165d

**Comment by bkerley** [comment]
@borshop merge

**Pull request merged**